internal/manager/manager.go

```diff
@@ -171,6 +171,37 @@ func (m *Manager) GetMetadata(inode uint64) (*common.Metadata, error) {
 	return md, nil
 }
 
+func (m *Manager) GetDirectoryContent(inode uint64) ([]common.Metadata, error) {
+	m.rLock()
+	defer m.rUnlock()
+
+	db, err := m.getDBClient()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't connect to database: %v", err)
+	}
+	defer db.Close()
+
+	md, err := db.Get(inode)
+	if err != nil {
+		if err == common.ErrNotFound {
+			return nil, err
+		}
+
+		return nil, fmt.Errorf("something went wrong with query: %v", err)
```

Comment:
Why not treat all errors equally? We don't handle `ErrNotFound` differently, we just log it explicitly which can still be done since `err` probably contains a sensible description.

Reply:
Actually, it will make sense when we implement fuse functions. There it will be transformed into ENOTDIR response.